src/compiler/Stage/InferTypes/AssignIds.elm

```diff
@@ -170,3 +170,21 @@ assignIdsWith idSource expr =
 
         Canonical.Unit ->
             wrap idSource Typed.Unit
+
+        Canonical.List list_ ->
```

Comment:
aand again `list_` -> `items` or `list`. Seeing as the rest of the function uses `items_`, I guess `items` makes most sense here.